lib/perl/Genome/Model/Tools/Manta/Base.pm

```diff
@@ -0,0 +1,246 @@
+}
+
+# tool subcommand name as a string (e.g., 'configManta.py')
+sub _tool_subcommand_name {
+    confess "sub _tool_subcommand_name must be overridden in child class";
+}
+
+sub _resolve_output_files {
+    # example:
+}
+
+# Input Files
+sub _resolve_input_files {
+    my $self = shift;
+
+    my @metas = $self->_tool_input_file_metas;
+
+    my @input_files = map { $self->_tool_input_files_from_meta($_) } @metas;
+    $self->input_files(\@input_files);
+    return @input_files;
+}
+
+# tool params should be 'params' and also have tool_arg_name defined
+sub _tool_param_metas {
+    my $class = shift;
+    my @param_property_metas = $class->__meta__->properties(is_param => 1);
+    return grep {$_->can("tool_arg_name") && $_->tool_arg_name} @param_property_metas;
+}
+
+# tool inputs should be 'inputs' and also have tool_arg_name defined
+sub _tool_input_metas {
+    my $class = shift;
+    my @input_property_metas = $class->__meta__->properties(is_input => 1);
+    return grep {$_->can("tool_arg_name") && $_->tool_arg_name} @input_property_metas;
+}
+
+# tool input files should be 'inputs' and also have tool_input_file defined
+sub _tool_input_file_metas {
+    my $class = shift;
+    my @property_metas = $class->__meta__->properties(is_input => 1);
+    return grep {$_->can("tool_input_file") && $_->tool_input_file} @property_metas;
+}
+
+# Handle conversion from boolean => 'true'/'false' and general tool arg
+# formatting (argname=argvalue).
+sub _format_tool_arg {
+    my ($type, $name, $value) = @_;
+    return '--'. $name if ($type eq 'Boolean');
+    $value = qq{"$value"} if $type eq 'Text';
+    return sprintf('--%s %s', $name, $value);
+}
+
+# given a property meta object (defining property name, and ostensibly
+# tool_arg_name), return the list of cmdline args that should be passed
+# to the tool based on the values set on the current object.
+#
+# this is where is_many properties get handled.
+sub _tool_args_from_meta {
+    my ($self, $meta) = @_;
+
+    my $ur_name = $meta->property_name;
+    # using // instead of ? : below made a lims perl test fail
+    my $type = defined $meta->data_type ? $meta->data_type : 'String';
+    my $tool_arg_name = $meta->tool_arg_name;
+
+    return unless defined $self->$ur_name;
+    if ($type eq 'Boolean' && $self->$ur_name eq '0') {
+        return;
+    }
+
+    # This works for things that are is_many or not.
+    return map {_format_tool_arg($type, $tool_arg_name, $_)} $self->$ur_name;
+}
+
+sub _tool_input_files_from_meta {
+    my ($self, $meta) = @_;
+
+    my $ur_name = $meta->property_name;
+
+    return unless defined $self->$ur_name;
+
+    # This works for things that are is_many or not.
+    return map {$_} $self->$ur_name;
+}
+
+# command line args that come after the executable
+#
+# The default implementation just translates all the inputs that have
+# speedseq_param_name defined. Boolean values get translated to true/false,
```

Comment:
This reference to `speedseq` seems like a hold-over from `Genome::Model::Tools::Speedseq::Base`.